src/Query/DateParser.ts

```diff
@@ -20,36 +20,54 @@ export class DateParser {
      * @return - A Tuple of dates. If both input dates are invalid, then both ouput dates will be invalid.
      */
     public static parseDateRange(input: string): [moment.Moment, moment.Moment] {
+        // Try parsing a relative date range like 'current month'
         const relativeDateRange = DateParser.parseRelativeDateRange(input);
         if (relativeDateRange !== undefined) {
             return relativeDateRange;
         }
 
-        // If relative date range was not parsed, fallback on absolute date range with chrono
-        return DateParser.parseAbsoluteDateRange(input);
+        // Try '2022-W10' otherwise
+        const specificDateRange = DateParser.parseSpecificDateRange(input);
+        if (specificDateRange !== undefined) {
+            return specificDateRange;
+        }
+
+        // If previous failed, fallback on absolute date range with chrono
+        const absoluteDateRange = DateParser.parseAbsoluteDateRange(input);
+        if (absoluteDateRange !== undefined) {
+            return absoluteDateRange;
+        }
+
+        // If nothing worked return and invalid date range
+        return [moment.invalid(), moment.invalid()];
     }
 
-    private static parseAbsoluteDateRange(input: string): [moment.Moment, moment.Moment] {
+    private static parseAbsoluteDateRange(input: string): [moment.Moment, moment.Moment] | undefined {
         const result = chrono.parse(input, undefined, {
             forwardDate: true,
         });
 
+        // Check chrono parsing
         if (result.length === 0) {
-            return [moment.invalid(), moment.invalid()];
+            return undefined;
         }
 
         const startDate = result[0].start;
         const endDate = result[1] && result[1].start ? result[1].start : startDate;
         const start = window.moment(startDate.date());
         const end = window.moment(endDate.date());
 
+        // Check momentjs parsing
+        if (!start.isValid() || !end.isValid()) {
+            return undefined;
+        }
```

Comment:
The tests parse, but I don't understand why this is not a change in behaviour - I cannot spot the original code that did the same thing.

Reply:
It wasn't there actually. There is no test that shows this behaviour, because this case is chrono parses but moment doesn't. I have put this just as a failsafe.

Reply:
I would prefer to remove it. It is a potential change in behaviour, not called for by tests. 

I am particularly suspicious of the OR.

Reply:
ok